Example code does not build

```cpp
#include <iostream>
#include "xtensor/xarray.hpp"
#include "xtensor/xio.hpp"

int main() {
	xt::xarray<double> arr1
	  {{1.0, 2.0, 3.0},
	   {2.0, 5.0, 7.0},
	   {2.0, 5.0, 7.0}};

	xt::xarray<double> arr2
	  {5.0, 6.0, 7.0};

	xt::xarray<double> res = xt::view(arr1, 1) + arr2;

	std::cout << res;
}
```

Failes to compile on ubuntu 16.04 with g++5.4 with

```
/usr/local/include/xtensor/xview.hpp:338:113: error: invalid initialization of non-const reference of type ‘xt::xarray_container<xt::uvector<double, std::allocator<double> >, (xt::layout_type)1, std::vector<long unsigned int> >&’ from an rvalue of type ‘xt::xarray_container<xt::uvector<double, std::allocator<double> >, (xt::layout_type)1, std::vector<long unsigned int> >’
```

Maybe someone else can confirm?

--- a/include/xtensor/xreducer.hpp
+++ b/include/xtensor/xreducer.hpp
@@ -399,7 +399,7 @@ namespace xt
     template <class... Args>
     inline auto xreducer<F, CT, X>::operator()(Args... args) const -> const_reference
     {
-        std::array<std::size_t, sizeof...(Args)> arg_array = {static_cast<std::size_t>(args)...};
+        std::array<std::size_t, sizeof...(Args)> arg_array = {{static_cast<std::size_t>(args)...}};
         return element(arg_array.cbegin(), arg_array.cend());
     }
 

--- a/include/xtensor/xview.hpp
+++ b/include/xtensor/xview.hpp
@@ -102,8 +102,10 @@ namespace xt
         static constexpr layout_type static_layout = layout_type::dynamic;
         static constexpr bool contiguous_layout = false;
 
-        template <class CTA, class... SL>
-        explicit xview(CTA&& e, SL&&... slices) noexcept;
+        // The argument FSL avoids the compiler to call this constructor
+        // instead of the copy constructor when sizeof...(SL) == 0.
+        template <class CTA, class FSL, class... SL>
+        explicit xview(CTA&& e, FSL&& first_slice, SL&&... slices) noexcept;
 
         template <class E>
         self_type& operator=(const xexpression<E>& e);
@@ -328,13 +330,14 @@ namespace xt
      * Users should not call directly this constructor but
      * use the view function instead.
      * @param e the xexpression to adapt
+     * @param first_slice the first slice describing the view
      * @param slices the slices list describing the view
      * @sa view
      */
     template <class CT, class... S>
-    template <class CTA, class... SL>
-    inline xview<CT, S...>::xview(CTA&& e, SL&&... slices) noexcept
-        : m_e(std::forward<CTA>(e)), m_slices(std::forward<SL>(slices)...),
+    template <class CTA, class FSL, class... SL>
+    inline xview<CT, S...>::xview(CTA&& e, FSL&& first_slice, SL&&... slices) noexcept
+        : m_e(std::forward<CTA>(e)), m_slices(std::forward<FSL>(first_slice), std::forward<SL>(slices)...),
           m_shape(make_sequence<shape_type>(m_e.dimension() - integral_count<S...>() + newaxis_count<S...>(), 0))
     {
         auto func = [](const auto& s) noexcept { return get_size(s); };

--- a/test/test_xview.cpp
+++ b/test/test_xview.cpp
@@ -150,6 +150,22 @@ namespace xt
         EXPECT_EQ(s, v.shape());
     }
 
+    TEST(xview, temporary_view)
+    {
+        xt::xarray<double> arr1
+        { { 1.0, 2.0, 3.0 },
+        { 2.0, 5.0, 7.0 },
+        { 2.0, 5.0, 7.0 } };
+
+        xt::xarray<double> arr2
+        { 5.0, 6.0, 7.0 };
+
+        xt::xarray<double> res = xt::view(arr1, 1) + arr2;
+        EXPECT_EQ(7., res(0));
+        EXPECT_EQ(11., res(1));
+        EXPECT_EQ(14., res(2));
+    }
+    
     TEST(xview, iterator)
     {
         view_shape_type shape = {2, 3, 4};

